### PR TITLE
Fix for #188

### PR DESCRIPTION
**Pull request by hessammehr**
Test case included.

### CODE REVIEW

File: test/dsp.jl [hunk 1]
```diff
@@ -58,3 +58,13 @@ a = [1., 2., 1., 2.]
 b = [1., 2., 3.]
 @test conv(a, b) ≈ [1., 4., 8., 10., 7., 6.]
 @test conv(complex.(a, ones(4)), complex(b)) ≈ complex.([1., 4., 8., 10., 7., 6.], [1., 3., 6., 6., 5., 3.])
+
+# Test for issue #188: deconv mutates inputs
+let b = [4.0, 2.0, 1.0]; a = [2.0, 1.0]
```
Comment (martinholters) [inline test/dsp.jl]: Can you guard the test with `if VERSION >= v"0.7.0-DEV.602"`? Prior to that, the functions from `Base` are used which contain this bug, too, so the test fails.

**Comment by martinholters** [comment]
I'm actually working on making the package usable on Julia 0.7 again, and I'd prefer to wait with merging this until it's testable.

**Comment by martinholters** [comment]
Um, I was hoping resolving conflicts would re-trigger CI. Apparently it doesn't, so I'll close and re-open.

**Comment by martinholters** [comment]
Thanks for the patience here!

**Comment by hessammehr** [comment]
No worries, glad to see this through. Thanks for all the good work.

P.S. I can try to lend a hand with future PRs if given commit rights.

**Pull request merged**